### ISSUE TITLE
Accept negative start line number

### ISSUE THREAD

**Issue by eregon** [ruby/prism#1783]
The start line can be negative and this is used in ERB, other template engines, etc so they can inject extra lines at the start without affecting the line number of the user source:
https://github.com/ruby/erb/blob/8db8b8b5086713175523f7cab4bc8b779f069709/lib/erb.rb#L468

See https://github.com/ruby/prism/pull/1763#discussion_r1384983631

**Comment by enebo** [comment]
:+1: 

```text
        // FIXME: I believe line number can be negative?
        // Line Number
        appendUnsignedInt(metadata, lineNumber);
```

**Comment by eregon** [comment]
I splitted the previous label in two because it wasn't clear.
It's intended to signal priority yes.
The meaning is that this issue is blocking correct compilation (of such eval's) and blocking truffleruby's adoption of prism (which is just another case of compilation, so maybe I should remove the `/truffleruby` suffix).

**Comment by eregon** [comment]
Yeah blocker was probably too strong without more context, I renamed to `needed for correct compilation`.
I'm trying to identify issues which do prevent switching from the current parser to Prism in TruffleRuby (by default, and removing the current parser).
I annotated the issues that I consider are blocking that, because it would be too incorrect or messy to work around (and since we are designing a new parser we don't want to make things difficult to use).
This specific issue I think is hard to work around so I think it qualifies for that (and it's likely easy to fix).

OTOH things like different warnings messages or missing SyntaxError I think are not very important for that switch, those could be fixed later from the POV of using Prism in TruffleRuby.
